docs/minio-as-eucalyptus-object-storage.md

```diff
@@ -0,0 +1,120 @@
+
+If everything goes well, when the services are enabled, we can now check if the supported object storage providers again.
+
+```
+# euctl -r objectstorage.providerclient
+euctl: error (ModifyPropertyValueType): Cannot modify \
+objectstorage.providerclient.providerclient new value \
+is not a valid value.  Legal values are: walrus,ceph-rgw,minio,riakcs
+```
+
+Looks like minio is now one of the supported provider client for Eucalyptus object storage. That was easy! :)
+
+At this point we should be able to set minio as a provider client for Eucalyptus object storage.
+
+```
+# euctl objectstorage.providerclient=minio
+objectstorage.providerclient = minio
+```
+
+Now configure Eucalyptus ObjectStorage service to use our Minio deployment endpoint (public IP address/hostname) and credentials,
+
+```
+# euctl objectstorage.s3provider.s3endpoint=10.116.159.114:9000
+# euctl objectstorage.s3provider.s3accesskey=GFQX5XMP1DSTIQ8JKRXN
+# euctl objectstorage.s3provider.s3secretkey=aFLQegoeIgzF/hK+Xymba7JwSANfn98ANNcKXz+S
+```
+
+Eucalyptus ObjectStorage gateway does a connectivity check with the provided S3 endpoint to enable the objectstorage service of Eucalyptus. Basically it expects a HTTP response code for HEAD operation on S3 endpoint, which can be configured based on the backend, as different backend can send different response code. Apparently Minio returns 404 on it's endpoint. So, to make it work, we need to set the value to 404 (feels weird though).
+
+```
+# euctl objectstorage.s3provider.s3endpointheadresponse=404
+```
+Within a few seconds we should see the object storage is enabled. We can run the following to check the service status of object storage,
+
+```
+# euserv-describe-services --filter service-type=objectstorage
+```
+
+Now we can use Minio as an object storage backend. However, while working, I found an issue where any PUT object request that was made through Eucalyptus ends up with a failure. After debugging with Swathi Gangisetty it looks like the AWS Java SDK Eucalyptus using doesn't like the fact that Minio is returning Etag instead of ETag as it is mentioned in the AWS Documentation. To test, we removed Eucalyptus out of the picture and tried to upload an object directly with AWS Java SDK 1.7.1, which also failed in the same way. However, the same upload request works with AWS Java SDK 1.10.x.
+So, to use Minio successfully as an Eucalyptus ObjectStorage backend, we either need a fix in Minio where it returns the correct ETag or when Eucalyptus starts using a newer AWS Java SDK in a future release.
+
+Here are the requests for the fix in Minio: https://github.com/minio/minio/issues/3284
+and for the AWS Java SDK update in Eucalyptus: https://eucalyptus.atlassian.net/browse/EUCA-12955
+
+Update:
+Looks like this was a bug in AWS Java SDK 1.7.1 and fixed in a later release,
+https://github.com/aws/aws-sdk-java/pull/590
```

Comment:
Since the issue is fixed, pls remove these details and mention the AWS SDK version that works.